#151948280 User should be able to configure periodic reports

kasulani opened the pull request:
#### What does this PR do?
* Adds a feature User should be able to configure periodic reports, for example monthly, weekly, daily.
#### Description of the task to be completed?
* The dashboard only allows subscribing to monthly report. This should be changed to be user configurable to either be daily, weekly or monthly.
#### How should this manually be tested?
* First run automatic tests from the project root directory and ensure they all pass; 
>python manage.py test hc.accounts.tests_profile 
* To manually test this PR; 
1. Log into the application
2. Navigate to the account settings button at the top right corner of your screen and click on it
3. Navigate to the periodic reports section and click on the configure button. This brings up a modal form where you can select the different reporting periods user can choose from
4. After selecting a reporting period click on the save button, to apply changes. See screenshots section for more details
#### Pivotal Tracker story?
* See [Feature #151948280](https://www.pivotaltracker.com/story/show/151948280)
#### Screenshots (if appropriate)
##### Step 1:
<img width="1260" alt="screen shot 2017-11-02 at 12 14 40" src="https://user-images.githubusercontent.com/6205925/32321860-a46cbb02-bfd3-11e7-830c-5115a8859637.png">

##### Step 2:
<img width="1205" alt="screen shot 2017-11-02 at 12 13 27" src="https://user-images.githubusercontent.com/6205925/32321893-b7b8cc3c-bfd3-11e7-9232-f9faaaec00af.png">

##### Step 3:
<img width="1214" alt="screen shot 2017-11-02 at 12 13 54" src="https://user-images.githubusercontent.com/6205925/32321912-c24fec8e-bfd3-11e7-9a41-23d921d39c67.png">

#### Questions:

kasulani committed:
refactors ReportSettingsForm adds report_period field

kasulani committed:
refactors ReportSettingsForm adds report_period field

kasulani committed:
adds field report_period to Profile model refactors send_report method

kasulani committed:
refactors profile controller

kasulani committed:
refactors profile controller

kasulani committed:
refactors profile controller

kasulani committed:
refactors profile controller

kasulani committed:
adds radio buttons to select report period

kasulani committed:
adds radio buttons to select report period

kasulani committed:
adds a test test_selected_periodic_report

kasulani committed:
adds a test test_selected_periodic_report

kasulani committed:
adds migration file

ivantoz commented:
- your PR heading and template does not follow our guidelines
- Your commit messages does not adhere to our standards
- Your PR description should include screenshots of the feature that you developed
- You should also provide manual description of how to test the feature that you delivered

jokamjohn committed:
feat(user can configure periodic reports):
- implements sending periodic reports ie daily, weekly, monthly
[Feature #151948280]

jokamjohn committed:
feat(user can configure periodic reports):
- Changes email body text depending on the period selected on profile ie daily, weekly, monthly
- Changes email subject depending on the period selected on profile ie daily, weekly, monthly
[Feature #151948280]

kasulani committed:
[Feature #151948280] adds reports_allowed field

kasulani committed:
[Feature #151948280] refactors profile view

kasulani committed:
[Feature #151948280] refactors profile view

kasulani committed:
[Feature #151948280] refactors profile view

kasulani committed:
[Feature #151948280] adds modal form

kasulani committed:
[Feature #151948280] adds modal form

kasulani committed:
[Feature #151948280] refactors profile model

jokamjohn committed:
[Feature #151948280] refactors the periodic reports cofiguration modal form

jokamjohn committed:
[Feature #151948280] refactors the periodic reports cofiguration modal form

jokamjohn committed:
[Feature #151948280] refactors the profile view

jokamjohn committed:
[Feature #151948280] refactors test_selected_periodic_report

jokamjohn committed:
test(profile) add tests for sending periodic reports
- ensure a user is sent the appropriate report ie. daily, weekly or monthly
- ensure the period selected by user is persisted in the database.
[Feature #151948280]

jokamjohn committed:
test(profile) add tests for sending periodic reports
- ensure a user is sent the appropriate report ie. daily, weekly or monthly
- ensure the period selected by user is persisted in the database.
[Feature #151948280]

kasulani committed:
refactors the profile view

kasulani committed:
refactors the periodic reports UI functionality

kasulani committed:
[Chore #151948277] add assert count on new profile object in TestMidd… …

kasulani committed:
[Feature #151948280] refactors periodic reporting section

kasulani committed:
removes conflicts

kasulani committed:
[Feature #151948280] removes print statement used in debugging

ivantoz commented on README.md:
```diff
@@ -1,3 +1,7 @@
+[![Build Status](https://travis-ci.org/andela/healthchecks-apollo.svg?branch=develop)](https://travis-ci.org/andela/healthchecks-apollo)
```
Why is your PR changing all this file?

ivantoz commented on Procfile:
```diff
@@ -0,0 +1 @@
+web: gunicorn hc.wsgi --log-file -
```
Why are you changing this file?

ivantoz commented on hc/accounts/models.py:
```diff
@@ -56,17 +57,23 @@ def set_api_key(self):
         # reset next report date first:
         now = timezone.now()
-        self.next_report_date = now + timedelta(days=30)
+        if self.report_period == 0:  # monthly report
```
Can you use enum here for the purpose of understanding the types of report period

kasulani replied:
We are fixing this

ivantoz commented on hc/accounts/models.py:
```diff
@@ -56,17 +57,23 @@ def set_api_key(self):
-        self.next_report_date = now + timedelta(days=30)
+        if self.report_period == 0:  # monthly report
+            self.next_report_date = now + timedelta(days=30)
+        elif self.report_period == 1:  # weekly report
```
same for this

jokamjohn replied:
working on it

ivantoz commented on hc/accounts/models.py:
```diff
@@ -56,17 +57,23 @@ def set_api_key(self):
+        elif self.report_period == 1:  # weekly report
+            self.next_report_date = now + timedelta(days=7)
+        elif self.report_period == 2:  # daily report
+            self.next_report_date = now + timedelta(days=1)
```
same here

jokamjohn replied:
Working on it

ivantoz commented on templates/accounts/profile.html:
```diff
@@ -5,323 +5,401 @@
+                                        <label>
+                                            <input
+                                                    name="report_period"
+                                                    type="radio"
```
Can you use django forms here for the purpose of learning

jokamjohn replied:
Working on it

kasulani committed:
adds enum for the report periods

kasulani committed:
adds enum

kasulani committed:
[Feature #151948280] adds ReportConfigForm

kasulani committed:
[Feature #151948280] refactors to use ReportConfigForm

kasulani committed:
[Feature #151948280] refactors to use Enum

kasulani committed:
[Feature #151948280] refactors to use ReportConfigForm passed in context

Commit:
Merge branch 'develop' into ft-user-configure-periodic-reports-151948280
